src/main/kotlin/no/nav/familie/ks/sak/integrasjon/økonomi/internkonsistensavstemming/InternKonsistensavstemmingUtil.kt

```diff
@@ -0,0 +1,110 @@
+import no.nav.familie.kontrakter.felles.oppdrag.Utbetalingsperiode
+import no.nav.familie.ks.sak.common.tidslinje.Periode
+import no.nav.familie.ks.sak.common.tidslinje.Tidslinje
+import no.nav.familie.ks.sak.common.tidslinje.tilTidslinje
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.kombinerMed
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.tilPerioder
+import no.nav.familie.ks.sak.common.util.førsteDagIInneværendeMåned
+import no.nav.familie.ks.sak.common.util.sisteDagIMåned
+import no.nav.familie.ks.sak.common.util.slåSammen
+import no.nav.familie.ks.sak.common.util.toLocalDate
+import no.nav.familie.ks.sak.common.util.toYearMonth
+import no.nav.familie.ks.sak.integrasjon.secureLogger
+import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelse
+import java.math.BigDecimal
+
+fun erForskjellMellomAndelerOgOppdrag(
+    andeler: List<AndelTilkjentYtelse>,
+    utbetalingsoppdrag: Utbetalingsoppdrag?,
+    fagsakId: Long
+): Boolean {
+    val utbetalingsperioder =
+        utbetalingsoppdrag?.utbetalingsperiode
+            ?.filter { it.opphør == null }
+            ?: emptyList()
+
+    val forskjellMellomAndeleneOgUtbetalingsoppdraget =
+        hentForskjellIAndelerOgUtbetalingsoppdrag(utbetalingsperioder, andeler)
+
+    when (forskjellMellomAndeleneOgUtbetalingsoppdraget) {
+        is UTBETALINGSPERIODER_UTEN_TILSVARENDE_ANDEL -> secureLogger.info(
+            "Fagsak $fagsakId har sendt utbetalingsperiode(r) til økonomi som ikke har tilsvarende andel tilkjent ytelse." +
+                "\nDet er differanse i perioden(e) ${forskjellMellomAndeleneOgUtbetalingsoppdraget.utbetalingsperioder.tilTidStrenger()}." +
+                "\n\nSiste utbetalingsoppdrag som er sendt til familie-øknonomi på fagsaken er:" +
+                "\n$utbetalingsoppdrag"
+        )
+
+        is INGEN_FORSKJELL -> Unit
+    }
+
+    return forskjellMellomAndeleneOgUtbetalingsoppdraget !is INGEN_FORSKJELL
+}
+
+private fun hentForskjellIAndelerOgUtbetalingsoppdrag(
+    utbetalingsperioder: List<Utbetalingsperiode>,
+    andeler: List<AndelTilkjentYtelse>
+): AndelOgOppdragForskjell {
+    val utbetalingsperioderUtenTilsvarendeAndel = utbetalingsperioder.filter {
+        it.erIngenPersonerMedTilsvarendeAndelITidsrommet(andeler)
+    }
+
+    return if (utbetalingsperioderUtenTilsvarendeAndel.isEmpty()) {
+        INGEN_FORSKJELL
+    } else {
+        UTBETALINGSPERIODER_UTEN_TILSVARENDE_ANDEL(utbetalingsperioderUtenTilsvarendeAndel)
+    }
+}
+
+private fun Utbetalingsperiode.erIngenPersonerMedTilsvarendeAndelITidsrommet(
+    andeler: List<AndelTilkjentYtelse>
+): Boolean {
+    val andelsTidslinjerPerPersonOgYtelsetype = andeler
+        .groupBy { Pair(it.aktør, it.type) }
+        .map { (_, andeler) -> andeler.tilBeløpstidslinje() }
+
+    return andelsTidslinjerPerPersonOgYtelsetype.all {
+        !this.harTilsvarendeAndelerForPersonOgYtelsetype(it)
+    }
```

Comment:
Dette er vel det samme, men lettere å lese?
```suggestion
    return andelsTidslinjerPerPersonOgYtelsetype. none {
        this.harTilsvarendeAndelerForPersonOgYtelsetype(it)
    }
```